### PR TITLE
Refactor request handling to activate API selection

### PR DESCRIPTION
**Pull request by canepat**
Configure the API handle methods separately
Make API Server build the handler table at startup
Add API specification command-line option
Fix integration test runner

Marked draft because unit tests still missing

### CODE REVIEW

File: silkrpc/http/request_handler.cpp [hunk 1]
```diff
@@ -24,86 +24,104 @@
...
-    {method::k_tg_getHeaderByHash, &commands::RpcApi::handle_tg_get_header_by_hash},
-    {method::k_tg_getHeaderByNumber, &commands::RpcApi::handle_tg_get_header_by_number},
-    {method::k_tg_getLogsByHash, &commands::RpcApi::handle_tg_get_logs_by_hash},
-    {method::k_tg_forks, &commands::RpcApi::handle_tg_forks},
-    {method::k_tg_issuance, &commands::RpcApi::handle_tg_issuance},
-    {method::k_parity_getBlockReceipts, &commands::RpcApi::handle_parity_get_block_receipts},
-};
+std::map<std::string, RequestHandler::HandleMethod> RequestHandler::handlers_;
+
+void RequestHandler::add_debug_handlers() {
```
Comment (Giulio2002) [inline silkrpc/http/request_handler.cpp:37]: We probably need to make `handlers_` non-static here

File: silkrpc/http/request_handler.cpp [hunk 2]
```diff
@@ -24,86 +24,104 @@
...
+    handlers_[method::k_net_listening] = &commands::RpcApi::handle_net_listening;
+    handlers_[method::k_net_peerCount] = &commands::RpcApi::handle_net_peer_count;
+    handlers_[method::k_net_version] = &commands::RpcApi::handle_net_version;
+}
+
+void RequestHandler::add_parity_handlers() {
+    handlers_[method::k_parity_getBlockReceipts] = &commands::RpcApi::handle_parity_get_block_receipts;
+}
+
+void RequestHandler::add_tg_handlers() {
```
Comment (Giulio2002) [inline silkrpc/http/request_handler.cpp:101]: should it not be `erigon_`? just curiosity, not relevant to the PR's scope

File: silkrpc/http/request_handler.cpp (hunk 1 above)
Comment (canepat) [reply silkrpc/http/request_handler.cpp:37]: Yes, you're right, we will in another PR. This is just a refactoring in preparation for that

File: silkrpc/http/request_handler.cpp (hunk 2 above)
Comment (canepat) [reply silkrpc/http/request_handler.cpp:101]: Yes, there's already an issue for that: see #101

**Review by Giulio2002 — APPROVED** [review]
I will leave a nice tick here :)

**Review by lupin012 — CHANGES_REQUESTED** [review]

File: silkrpc/http/server.cpp [hunk 3]
```diff
@@ -30,19 +31,74 @@
...
+    return {host, port};
+}
+
+std::vector<std::string> Server::parse_api_spec(const std::string& api_spec) {
+    std::vector<std::string> api_namespaces;
+
+    auto start = 0u;
+    auto end = api_spec.find(kApiSpecSeparator);
+    while (end != std::string::npos) {
+        api_namespaces.push_back(api_spec.substr(start, end - start));
```
[the remark this answers is not in the document]
Comment (canepat) [reply silkrpc/http/server.cpp]: Sure, you're right

**Review by lupin012 — APPROVED** [review]

**Pull request merged**